In [12]:
import math

import math
import matplotlib.pyplot as pl

a,b,fo,f1,xo=[],[],[],[],[]
a=[[0.5,-1],[1.02,-2]]
b=[-9.5,-18.8]
x1,x2,a11=0,0,-0.5

def inicializar_Matriz(fil,col):
    
    mat= [None] * fil
    
    if(fil>0):
        for i in range(fil):
            mat[i] = [None] * col
            
    return mat

def func1_X1(x): 
    y=(-9.5 +x)/a11
    return y

def func1_X2(x): 
    y=(a11*x +9.5)
    return y

def func2_X1(x): 
    y=(2*x -18.8)/1.02
    return y

def func2_X2(x): 
    y=(1.02*x +18.8)/2
    return y

def ecua1(x1,x2):
    
    y=(a11*x1)-(x2)
    return y

def ecua2(x1,x2):
    
    y=1.02*x1 -2*x2
    return y


def metodo_grafico():
    
    global x1,x2
    n,i=0,0
    
    while(i<500):
        
        if(func1_X1(n)==func2_X1(n)):

            x1=func1_X1(n)
            x2=func2_X2(x1)
            
        fo.append(func1_X1(n))
        f1.append(func2_X1(n))
        xo.append(n)
        n+=0.1
        i+=1
    
    if(x1!=0 and x2!=0):
        print("x1=",x1)
        print("x2=",x2)
        print("(0.5*x1)-x2 =",ecua1(x1,x2))
        print("(1.02*x1)-2*x2 =",ecua2(x1,x2))
        
def graf():
    
    pl.ylabel('X1')
    pl.xlabel('X2')
    
    pl.plot(x2,x1,"ro",color="black",linewidth=20)
    pl.plot(xo,fo,color="green",label="Primera Ecuacion",linewidth=2.5, linestyle="-")
    pl.plot(xo,f1,color="pink",label="Segunda Ecuacion",linewidth=2.5, linestyle="-")
   
    pl.axis([6,30,0,15])
    pl.grid(True)
    pl.show()

def cal_determinante_dos_por_dos(a):
    return (a[0][0]*a[1][1])-(a[0][1]*a[1][0])

def eliminacion_incg(a,b):
    
    global x1,x2
    mat=inicializar_Matriz(len(a),len(a))
    sum=inicializar_Matriz(len(a)+1,0)
    
    cons,aux,auxx,i,j,x,z,cf=0,0,0,0,0,0,0,(len(a))

    while i<len(mat):
        x-=1
        cf=a[x][0]
        while j<len(mat[0]):
            mat[i][j]=a[i][j]*cf
            j+=1
        
        j=0
        
        cons=b[i]
        
        if(i>0):
            auxx=auxx-(cf*cons)
        else:
            auxx=auxx+(cf*cons)

        j=0
        i+=1
    
    sum[0]=mat[0][0]-mat[1][0]
    sum[1]=mat[0][1]-mat[1][1]
    sum[i]=(auxx)
    
    x2=(sum[2]/sum[1])
    x1=func2_X1(x2)
    print(mat,sum)
    print("x1=",x1)
    print("x2=",x2)
    print("(0.5*x1)-x2 =",ecua1(x1,x2))
    print("(1.02*x1)-2*x2 =",ecua2(x1,x2))
    
def modificar_mat():

    global a,a11
    a[0][0]=0.52
    a11=0.52
        
metodo_grafico()
print("Determinanre_A:",cal_determinante_dos_por_dos(a))
graf()
eliminacion_incg(a,b)
modificar_mat()
eliminacion_incg(a,b)

Determinanre_A: 0.020000000000000018
[[0.51, -1.02], [0.51, -1.0]] [0.0, -0.020000000000000018, -0.28999999999999915]
x1= 9.999999999999892
x2= 14.499999999999945
(0.5*x1)-x2 = -19.49999999999989
(1.02*x1)-2*x2 = -18.8
[[0.5304, -1.02], [0.5304, -1.04]] [0.0, 0.020000000000000018, 0.08600000000000207]
x1= -9.999999999999805
x2= 4.3000000000001
(0.5*x1)-x2 = -9.5
(1.02*x1)-2*x2 = -18.8


9.7 Dadas las ecuaciones
0.5x1 – x2 = –9.5
1.02x1 – 2x2 = –18.8
a) Resuelva en forma gráfica.
b) Calcule el determinante.
c) Con base en los incisos a) y b), ¿qué es de esperarse con
respecto de la condición del sistema?
d) Resuelva por medio de la eliminación de incógnitas.
e) Resuelva otra vez, pero modifique ligeramente el elemento
a11 a 0.52. Interprete sus resultados.

Solucion

a)
x1= 9.999999999999929
x2= 14.499999999999964
(0.5*x1)-x2 = -9.5
(1.02*x1)-2*x2 = -18.8
b) Determinanre_A: 0.020000000000000018
c)El sistema esta mal condicionado las pendientes son muy cercanas, el determinante es muy pequeño 
d)
x1= 9.999999999999892
x2= 14.499999999999945
(0.5*x1)-x2 = -19.49999999999989
(1.02*x1)-2*x2 = -18.8
e)
x1= -9.999999999999805
x2= 4.3000000000001
Un pequeño cambio en uno de los coeficientes modifica en gran medida los valores X1 y X2